kore/src/Kore/Repl/Interpreter.hs

```diff
@@ -992,6 +1004,75 @@ saveSession path =
     seqUnlines :: Seq String -> String
     seqUnlines = unlines . toList
 
+savePartialProof
+    :: forall m claim
+    .  MonadState (ReplState claim) m
+    => MonadWriter ReplOutput m
+    => MonadIO m
+    => Claim claim
+    => From claim (TermLike Variable)
+    => Maybe Natural
+    -> FilePath
+    -> m ()
+savePartialProof maybeNatural file = do
+    currentClaim <- Lens.use (field @"claim")
+    currentIndex <- Lens.use (field @"claimIndex")
+    claims <- Lens.use (field @"claims")
+    maybeConfig <- getConfigAt maybeNode
+    case maybeConfig of
+        Nothing -> putStrLn' "Invalid node!"
+        Just (_, currentProofState) -> do
+            let config = unwrapConfig currentProofState
+                newClaim = createClaim currentClaim config
+                newTrustedClaims =
+                    makeTrusted <$> removeAt currentIndex claims
```

Comment:
If this removes the current claim: except before the first step, the current claim should also be marked as trusted (in coinductive proofs, the claim being proven can be use after the first step).

Reply:
That's right, thanks for catching this!